droopler.install

```diff
@@ -1818,3 +1822,86 @@ function droopler_update_8127(&$sandbox) {
 
 }
 
+/**
+ * Droopler 2.2 Migrate layout settings.
+ */
+function droopler_update_8128(&$sandbox) {
+
+  $bundles = [
+    'd_p_tiles',
+    'd_p_side_tiles',
+    'd_p_side_image',
+    'd_p_side_embed',
+    'd_p_side_form',
+  ];
+  $paragraphs_per_batch = 25;
+  if (!isset($sandbox['current'])) {
+    $sandbox['current'] = 0;
+  }
+
+  $paragraphs_ids = \Drupal::entityQuery('paragraph')
+    ->condition('type', $bundles, 'IN')
+    ->range($sandbox['current'], $paragraphs_per_batch)
+    ->execute();
+
+  foreach ($paragraphs_ids as $id) {
+    $paragraph = \Drupal::entityTypeManager()
+      ->getStorage('paragraph')
+      ->load($id);
+
+    $settings = $paragraph->hasField('field_d_settings') && $paragraph->get('field_d_settings') ? $paragraph->get('field_d_settings')->value : "{}";
+    $json = json_decode($settings);
+
+    switch ($paragraph->bundle()) {
+      case 'd_p_side_image': case 'd_p_side_embed':
+          $old_settings = $paragraph->hasField('field_d_p_si_image_side') && $paragraph->get('field_d_p_si_image_side') ? $paragraph->get('field_d_p_si_image_side')->value : [];
+        break;
+
+      case 'd_p_side_form':
+        $old_settings = $paragraph->hasField('field_d_layout_modifier') && $paragraph->get('field_d_layout_modifier') ? $paragraph->get('field_d_layout_modifier')->value : [];
+        break;
+
+      case 'd_p_side_tiles': case 'd_p_tiles':
```

Comment:
oo to ja jeszcze tu jedna prośbę mam: nie moglibysmy tego mieć jeden pod drugim ?

Reply:
aha, jeszcze inaczej, te casy tytaj raczej nie powinny byc "scalone"

`d_p_tiles` nie ma layoutu, 
a `d_p_side_tiles` nie maja featured images :thinking: